torch/_refs/__init__.py

```diff
@@ -1471,6 +1471,26 @@ def logaddexp(a: TensorLikeType, b: TensorLikeType) -> TensorLikeType:
         return torch.where(inf_mask, a, max_ + torch.log1p(torch.exp(min_ - max_)))
 
 
+@_make_elementwise_binary_reference(
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+    supports_lhs_python_scalar=False,
+    supports_rhs_python_scalar=False,
+)
+def logaddexp2(a: TensorLikeType, b: TensorLikeType) -> TensorLikeType:
+    utils.check(
+        not (utils.is_complex_dtype(a.dtype) or utils.is_complex_dtype(b.dtype)),
+        lambda: "logaddexp2 doesn't support complex dtypes",
+    )
+    # Nb. this implementation does not distribute the gradients evenly when a == b
```

Comment:
Is this a divergence from eager? Or this doesn't matter since we're applying it after autograd?

Reply:
tbh I just copied the comment from logaddexp, I agree it shouldn't matter here since it gets decomposed after autograd.